File: array_coloring/array.py
def main():
    t = int(input())
    for _ in range(t):
        n = input()
        arr = list(map(int, input().split()))
        if (arr[0] % 2 == sum(arr[1:]) % 2):
            print("YES")
        else:
            print("NO")
    
    
if __name__ == "__main__":
    main()

File: array_coloring/array.cpp
#include <iostream>

using namespace std;

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n;
        cin >> n;
        int first;
        cin >> first;
        int sum = 0;
        int num;
        for (int j = 1; j < n; j++)
        {
            cin >> num;
            sum += num;
        }

        if ((first % 2) == (sum % 2))
        {
            cout << "YES\n";
        }

        else
        {
            cout << "NO\n";
        }
    }
}
